### PR TITLE
8296718: Refactor bootstrap Test Common Functionalities to test/lib/Utils

### PR DESCRIPTION
**Pull request by bwhuang-us**
There is no new changes in the open portion of JDK, but extracting common functionalities in bootstrap Utils to test/lib/Utils which can be used in the closed part.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8296718](https://bugs.openjdk.org/browse/JDK-8296718): Refactor bootstrap Test Common Functionalities to test/lib/Utils

### Reviewers
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**)
 * [Kevin Walls](https://openjdk.org/census#kevinw) (@kevinjwalls - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11029/head:pull/11029` \
`$ git checkout pull/11029`

Update a local copy of the PR: \
`$ git checkout pull/11029` \
`$ git pull https://git.openjdk.org/jdk pull/11029/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11029`

View PR using the GUI difftool: \
`$ git pr show -t 11029`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11029.diff">https://git.openjdk.org/jdk/pull/11029.diff</a>

</details>

### CODE REVIEW

**Comment by sspitsyn** [comment]
The refactoring looks pretty good.
But I have a couple of questions.
Q1: The method `grantAccess` is converted to the util method `grantFileAccess` and it does not look the same. Could you, please, explain what are the changes?
Q2: I'm confused about conversion of the shell tests. Your fix does not convert any shell test. Also, I do not see any changes to the shell ile `test/jdk/sun/management/jmxremote/bootstrap/ReadWriteForbidsCreateTest.sh` listed in the enhancement. Can you explain, please?

**Comment by bwhuang-us** [comment]
Hi @sspitsyn, the methods `grantAccess` and `grantFileAccess` are almost identical except for the parameter `AccessControl access` being changed to a boolean which is used to indicate user access or full access. There is a minor change in building a new ACL entry out of the original entry.

For your Q2 , it is in closed part of JDK. Please see the JBS ticket for the PR link.

**Review by sspitsyn — COMMENTED** [review]
I've placed a couple of comments.
Otherwise, it looks okay to me.
Thanks,
Serguei

File: test/lib/jdk/test/lib/Utils.java [hunk 1]
```diff
@@ -818,4 +834,200 @@ public static Path createTempDirectory(String prefix, FileAttribute<?>... attrs)
...
+                } else {
+                    if (thisEntry.type() != AclEntryType.ALLOW) {
+                        acl.add(AclEntry.newBuilder(thisEntry)
+                                .setType(AclEntryType.ALLOW)
+                                .build());
+                    } else {
+                        acl.add(thisEntry);
+                    }
+                }
+            }
```
Comment (sspitsyn) [inline test/lib/jdk/test/lib/Utils.java:1007]: What was your motivation to get rid of the Util methods revokeAccess, allowAccess and buildAclEntry.
The implementation did not become more clearer nor simpler. But reviewing it became much harder.

Comment (bwhuang-us) [reply test/lib/jdk/test/lib/Utils.java:1007]: Sure, I will add those methods back.

**Review by sspitsyn — APPROVED** [review]
Thank you for the update.
Looks good.
Thanks,
Serguei

**Review by kevinjwalls — APPROVED** [review]
Looks ok! 8-)

**Comment by bwhuang-us** [comment]
It seems there is an integration issue with this PR as the JBS ticket is marked as confidential. I will open a new ticket for this change.

**Pull request closed without merging**